engine/src/exec-strings.cpp

```diff
@@ -1227,9 +1232,19 @@ void MCStringsEvalFormat(MCExecContext& ctxt, MCStringRef p_format, MCValueRef*
                         {
                             // AL-2014-11-19: [[ Bug 14059 ]] Pad with zeroes if the appropriate specifier flag was used
                             if (t_zero_pad)
-                                t_success = MCStringAppendFormat(*t_result, "%0*s%@", width - t_range . length, "", *t_string);
+                            {
+                                if (!t_is_negative)
+                                    t_success = MCStringAppendFormat(*t_result, "%0*s%@", width - t_range . length, "", *t_string);
+                                else
+                                    t_success = MCStringAppendFormat(*t_result, "%@%0*s", *t_string, width - t_range . length, "");
```

Comment:
Is this definitely the LC6 behavior? i.e. if 0 padding is specified, and -ve (i.e. left alignment in width) then 0's appear after the number (thus changing the number printed...).

Reply:
Indeed, in the case of 0 padding this will not match the LC 6 behavior. I have just pushed an update to fix this use-case too:

LC 6: `put 6 into n1;put format ("%-05s", n1)` --> `6`
LC 7+: `put 6 into n1;put format ("%-05s", n1)` --> `00006`
this patch: `put 6 into n1;put format ("%-05s", n1)` --> `6`